3 o voting function

Lenopcha opened the pull request.

Lenopcha committed:
Implement the majority voting function major(x, y, z) (denoted as ⟨x,y,z⟩). The function returns the value that occurs most frequently among its arguments. The function can be defined by the following table:

 x | y | z | ⟨x,y,z⟩
---|---|---|--------
 0 | 0 | 0 |    0
 0 | 0 | 1 |    0
 0 | 1 | 0 |    0
 0 | 1 | 1 |    1
 1 | 0 | 0 |    0
 1 | 0 | 1 |    1
 1 | 1 | 0 |    1
 1 | 1 | 1 |    1

Input/output example:

major(False, True, False) | False

Lenopcha committed:
Implement the majority voting function major(x, y, z) (denoted as ⟨x,y,z⟩). The function returns the value that occurs most frequently among its arguments. The function can be defined by the following table:

 x | y | z | ⟨x,y,z⟩
---|---|---|--------
 0 | 0 | 0 |    0
 0 | 0 | 1 |    0
 0 | 1 | 0 |    0
 0 | 1 | 1 |    1
 1 | 0 | 0 |    0
 1 | 0 | 1 |    1
 1 | 1 | 0 |    1
 1 | 1 | 1 |    1

Input/output example:

major(False, True, False) | False

svichka commented on 3_O Voting Function/3-0 Voting Function more interactive.py:
```diff
@@ -0,0 +1,47 @@
+    print(f"Number of True values: {true_count}")
+
+    # print the result
+    if true_count >= 2:
```
Let's do it more general, without using hardcoded numbers

svichka replied:
Also we don't need `else` condition here

svichka commented on 3_O Voting Function/3_O Voting Function.py:
```diff
@@ -0,0 +1,16 @@
+    false_count = 3 - true_count
+
+    # Return the value that occurs most frequently
+    if true_count > false_count:
```
remove the `else` condition

svichka commented on 3_O Voting Function/3_O Voting Function.py:
```diff
@@ -0,0 +1,16 @@
+def major(x, y, z):
+    # Count the occurrences of True and False
+    true_count = sum([x, y, z])
+    false_count = 3 - true_count
```
Let's do it more general, without using hardcoded numbers

Lenopcha committed:
Implement the majority voting function major(x, y, z) (denoted as ⟨x,y,z⟩). The function returns the value that occurs most frequently among its arguments. The function can be defined by the following table:

 x | y | z | ⟨x,y,z⟩
---|---|---|--------
 0 | 0 | 0 |    0
 0 | 0 | 1 |    0
 0 | 1 | 0 |    0
 0 | 1 | 1 |    1
 1 | 0 | 0 |    0
 1 | 0 | 1 |    1
 1 | 1 | 0 |    1
 1 | 1 | 1 |    1

Input/output example:

major(False, True, False) | False

Lenopcha committed:
Implement the majority voting function major(x, y, z) (denoted as ⟨x,y,z⟩). The function returns the value that occurs most frequently among its arguments. The function can be defined by the following table:

 x | y | z | ⟨x,y,z⟩
---|---|---|--------
 0 | 0 | 0 |    0
 0 | 0 | 1 |    0
 0 | 1 | 0 |    0
 0 | 1 | 1 |    1
 1 | 0 | 0 |    0
 1 | 0 | 1 |    1
 1 | 1 | 0 |    1
 1 | 1 | 1 |    1

Input/output example:

major(False, True, False) | False

svichka commented on 3_O Voting Function/3-0 Voting Function more interactive.py:
```diff
@@ -0,0 +1,47 @@
+    print(f"Number of True values: {true_count}")
+
+    # Determine and print the result
+    result = true_count > len(values) // 2
```
let's calculate `values.count(True)` and `values.count(False)` and compare them

Lenopcha committed:
Print a Christmas Tree with input

Write a function fir(a, b) that takes two integers 1≤a≤b as input and prints a Christmas tree composed of b−a+1 triangles with heights a, a+1, ..., b. Use the triangle function from the previous task, modifying it to receive a second parameter: an additional number of spaces to be added at the beginning of each line of the displayed triangle. The bottom line of the tree should not contain spaces at the beginning. Submit two functions for review.

Lenopcha committed:
Voting Function _ new _
Implement the majority voting function major(x, y, z) (denoted as ⟨x,y,z⟩). The function returns the value that occurs most frequently among its arguments
